libcovebods/tasks/checks.py

```diff
@@ -695,3 +757,1164 @@ def check_entity_statement_first_pass(self, statement):
+                            }
+                        )
+
+
+class CheckStatementEntityFoundationDissolutionDates(AdditionalCheck):
+    def __init__(self, lib_cove_bods_config, schema_object):
+        super().__init__(lib_cove_bods_config, schema_object)
+
+    @staticmethod
+    def does_apply_to_schema(lib_cove_bods_config, schema_object) -> bool:
+        return schema_object.is_schema_version_equal_to_or_greater_than("0.4")
+
+    def check_entity_statement_first_pass(self, statement):
+        if (
+            "recordDetails" in statement
+            and isinstance(statement["recordDetails"], dict)
+            and "foundingDate" in statement["recordDetails"]
+            and statement["recordDetails"]["foundingDate"]
+            and "dissolutionDate" in statement["recordDetails"]
+            and statement["recordDetails"]["dissolutionDate"]
+        ):
+            founding_date = parse_date_field(statement["recordDetails"]["foundingDate"])
+            dissolution_date = parse_date_field(
+                statement["recordDetails"]["dissolutionDate"]
+            )
+            if founding_date > dissolution_date:
+                self._additional_check_results.append(
+                    {
+                        "type": "statement_entity_dissolution_before_founding_date",
+                        "statement_type": None,
+                        "founding_date": founding_date.strftime("%Y-%m-%d"),
+                        "dissolution_date": dissolution_date.strftime("%Y-%m-%d"),
+                        "statement": statement.get("statementId"),
+                    }
+                )
+
+
+class CheckStatementPersonBirthDateSensible(AdditionalCheck):
+    def __init__(self, lib_cove_bods_config, schema_object):
+        super().__init__(lib_cove_bods_config, schema_object)
+
+    @staticmethod
+    def does_apply_to_schema(lib_cove_bods_config, schema_object) -> bool:
+        return schema_object.is_schema_version_equal_to_or_greater_than("0.4")
+
+    def check_person_statement_first_pass(self, statement):
+        if (
+            "recordDetails" in statement
+            and isinstance(statement["recordDetails"], dict)
+            and "birthDate" in statement["recordDetails"]
+            and statement["recordDetails"]["birthDate"]
+        ):
+            birth_date = parse_date_field(statement["recordDetails"]["birthDate"])
+            if birth_date:
+                if birth_date > datetime.now().date():
+                    self._additional_check_results.append(
+                        {
+                            "type": "statement_person_birth_date_in_future",
+                            "statement_type": None,
+                            "birth_date": birth_date.strftime("%Y-%m-%d"),
+                            "statement": statement.get("statementId"),
+                        }
+                    )
+                elif birth_date < datetime.strptime("1800-01-01", "%Y-%m-%d").date():
```

Comment:
The config has bods_additional_checks_person_birthdate_min_year - could this use that?

Reply:
Have transitioned to using min date from config